src/web/api.rs

```diff
@@ -279,6 +285,112 @@ impl ApiService {
             session_id: self.serve_session.session_id(),
         })
     }
+
+    fn handle_api_create_assets(&self, request: Request<Body>) -> <Self as Service>::Future {
+        let session_id = self.serve_session.session_id();
+
+        let serve_tree = self.serve_session.tree_handle();
+        let vfs = self.serve_session.vfs_handle();
+
+        Box::new(request.into_body().concat2().and_then(move |body| {
+            let serve_tree = serve_tree.lock().expect("Couldn't lock RojoTree mutex");
+
+            let request: CreateAssetsRequest = match serde_json::from_slice(&body) {
+                Ok(request) => request,
+                Err(err) => {
+                    return json(
+                        ErrorResponse::bad_request(format!("Invalid body: {}", err)),
+                        StatusCode::BAD_REQUEST,
+                    );
+                }
+            };
+
+            if request.session_id != session_id {
+                return json(
+                    ErrorResponse::bad_request("Wrong session ID"),
+                    StatusCode::BAD_REQUEST,
+                );
+            }
+
+            let serve_dom = serve_tree.inner();
+
+            let packed_uuid = Uuid::new_v4().to_simple().to_string();
+            let model_path = format!("{}.rbxm", packed_uuid);
+
+            let studio = match RobloxStudio::locate() {
+                Ok(studio) => studio,
+                Err(error) => {
+                    return json(
+                        ErrorResponse::bad_request(format!(
+                            "Couldn't find Roblox Studio: {}",
+                            error
+                        )),
+                        StatusCode::INTERNAL_SERVER_ERROR,
+                    );
+                }
+            };
+
+            // TODO: memofs does not have create_dir_all, but it is necessary here.
```

Comment:
We don't really want to use memofs for stuff outside the file tree that we want cached, anyways.

Reply:
I thought it was for testability, so just use the filesystem directly?